apps/ukvi-complaints/behaviours/save-form-session.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const { model: Model } = require('hof');
+const config = require('../../../config');
+const { generateErrorMsg } = require('../../../lib/utils');
+const { protocol, host, port } = config.saveService;
+const applicationsUrl = `${protocol}://${host}:${port}/submitted_applications`;
+
+module.exports = superclass => class extends superclass {
+  requestBody(id, patchObj, postObj) {
+    if (id === undefined || id.length === 0) {
+      postObj.submitted_at = new Date();
+      return {
+        url: applicationsUrl,
+        method: 'POST',
+        data: postObj
+      };
+    }
+
+    // set submitted_at to current time
+    patchObj.submitted_at = new Date();
+    return {
+      url: applicationsUrl + `/${id}`,
+      method: 'PATCH',
+      data: patchObj
+    };
+  }
+
+  getSession(req) {
+    // remove csrf secret and errors from session data to prevent CSRF Secret issues in the session
+    const session = req.sessionModel.toJSON();
+    delete session['csrf-secret'];
+    delete session.errors;
+
+    if (session.steps.indexOf(req.path) === -1) {
+      session.steps.push(req.path);
+    }
+    // ensure no /edit steps are add to the steps property when we save to the store
+    session.steps = session.steps.filter(step => !step.match(/\/change|edit$/));
+
+    return session;
+  }
+
+  async saveValues(req, res, next) {
+    return super.saveValues(req, res, async err => {
+      if (err) {
+        return next(err);
+      }
+      const session = this.getSession(req);
+
+      // skip requesting data service api when running in local and test mode
+      if (config.env === 'local' || config.env === 'test') {
+        return next();
+      }
+
+      const id = req.sessionModel.get('id');
+      const submissionReference = req.sessionModel.get('submission-reference');
+      const params = this.requestBody(id, { session }, { submission_reference: submissionReference, session });
```

Comment:
this can be simplified, there is no need to to pass two similar objects, as you rely on `id` only
`submission_reference` can be included in both cases 'PATCH' and 'POST', as it will be present at all time

```suggestion
      const params = this.requestBody(id, { submission_reference: submissionReference, session });
```

Reply:
Done